### ISSUE TITLE
GCS - NPM dependencies

### ISSUE THREAD

**Issue by lazharichir** [googleapis/nodejs-storage#207]
Using GCS in Google Cloud Functions, the number of dependencies matters hugely in order to improve cold starts. I use GCS for one short upload (logging) and it seems like it is hurting my cold start times as it has over a hundred dependencies in total.
```
+ @google-cloud/storage@1.7.0
added 106 packages from 121 contributors in 7.289s
```
Are you folks caching your own packages for faster loading or not really?

**Comment by lazharichir** [comment]
Axios? No dependencies and does the job rather well... Thanks for the
update!

**Comment by JustinBeckwith** [comment]
Yes!  We've already moved a few core dependencies over, and it's gone quite well.  It's just a lot of work is all.

**Comment by fhinkel** [comment]
Thanks for raising the issue. Closing this as duplicate of https://github.com/googleapis/nodejs-storage/issues/207

**Comment by fhinkel** [comment]
@JustinBeckwith , why did you re-open this?

**Comment by JustinBeckwith** [comment]
We still have a dependency on request via gcs-resumable-upload.  Can you post the results of a npm ls --production?

**Comment by JustinBeckwith** [comment]
Also worth posting a trace using require-so-slow now and from the 1.x release.

**Comment by JustinBeckwith** [comment]
This was fixed in #619  :)   On master as of now:

```
@google-cloud/storage@2.4.2 /Users/beckwith/Code/nodejs-storage
├─┬ @google-cloud/common@0.31.0
│ ├── @google-cloud/projectify@0.3.2
│ ├── @google-cloud/promisify@0.3.1
│ ├─┬ @types/duplexify@3.6.0
│ │ └── @types/node@11.10.5
│ ├─┬ @types/request@2.48.1
│ │ ├── @types/caseless@0.12.1
│ │ ├─┬ @types/form-data@2.2.1
│ │ │ └── @types/node@11.10.5 deduped
│ │ ├── @types/node@11.10.5 deduped
│ │ └── @types/tough-cookie@2.3.5
│ ├── arrify@1.0.1 deduped
│ ├── duplexify@3.7.1 deduped
│ ├── ent@2.2.0
│ ├── extend@3.0.2 deduped
│ ├─┬ google-auth-library@3.1.0
│ │ ├── base64-js@1.3.0
│ │ ├── fast-text-encoding@1.0.0
│ │ ├── gaxios@1.8.1 deduped
│ │ ├─┬ gcp-metadata@0.9.3
│ │ │ ├── gaxios@1.8.1 deduped
│ │ │ └─┬ json-bigint@0.3.0
│ │ │   └── bignumber.js@7.2.1
│ │ ├─┬ gtoken@2.3.2
│ │ │ ├── gaxios@1.8.1 deduped
│ │ │ ├─┬ google-p12-pem@1.0.3
│ │ │ │ ├── node-forge@0.7.6
│ │ │ │ └── pify@4.0.1 deduped
│ │ │ ├── jws@3.2.1 deduped
│ │ │ ├── mime@2.4.0 deduped
│ │ │ └── pify@4.0.1 deduped
│ │ ├── https-proxy-agent@2.2.1 deduped
│ │ ├─┬ jws@3.2.1
│ │ │ ├─┬ jwa@1.4.0
│ │ │ │ ├── buffer-equal-constant-time@1.0.1
│ │ │ │ ├─┬ ecdsa-sig-formatter@1.0.11
│ │ │ │ │ └── safe-buffer@5.1.2 deduped
│ │ │ │ └── safe-buffer@5.1.2 deduped
│ │ │ └── safe-buffer@5.1.2 deduped
│ │ ├─┬ lru-cache@5.1.1
│ │ │ └── yallist@3.0.3
│ │ └── semver@5.6.0
│ ├── pify@4.0.1
│ └─┬ retry-request@4.0.0
│   └─┬ through2@2.0.5
│     ├── readable-stream@2.3.6 deduped
│     └── xtend@4.0.1 deduped
├─┬ @google-cloud/paginator@0.1.2
│ ├── arrify@1.0.1 deduped
│ ├── extend@3.0.2 deduped
│ ├── is@3.3.0
│ ├─┬ split-array-stream@2.0.0
│ │ └── is-stream-ended@0.1.4
│ └── stream-events@1.0.5 deduped
├── @google-cloud/promisify@0.4.0
├── arrify@1.0.1
├─┬ async@2.6.2
│ └── lodash@4.17.11
├─┬ compressible@2.0.16
│ └── mime-db@1.38.0
├─┬ concat-stream@2.0.0
│ ├── buffer-from@1.1.1
│ ├── inherits@2.0.3
│ ├─┬ readable-stream@3.2.0
│ │ ├── inherits@2.0.3 deduped
│ │ ├─┬ string_decoder@1.1.1
│ │ │ └── safe-buffer@5.1.2 deduped
│ │ └── util-deprecate@1.0.2
│ └── typedarray@0.0.6
├─┬ duplexify@3.7.1
│ ├─┬ end-of-stream@1.4.1
│ │ └── once@1.4.0 deduped
│ ├── inherits@2.0.3 deduped
│ ├─┬ readable-stream@2.3.6
│ │ ├── core-util-is@1.0.2
│ │ ├── inherits@2.0.3 deduped
│ │ ├── isarray@1.0.0
│ │ ├── process-nextick-args@2.0.0
│ │ ├── safe-buffer@5.1.2
│ │ ├── string_decoder@1.1.1 deduped
│ │ └── util-deprecate@1.0.2 deduped
│ └── stream-shift@1.0.0
├── extend@3.0.2
├─┬ gcs-resumable-upload@1.0.0
│ ├─┬ abort-controller@2.0.2
│ │ └── event-target-shim@5.0.1
│ ├─┬ configstore@4.0.0
│ │ ├─┬ dot-prop@4.2.0
│ │ │ └── is-obj@1.0.1
│ │ ├── graceful-fs@4.1.15
│ │ ├─┬ make-dir@1.3.0
│ │ │ └── pify@3.0.0
│ │ ├─┬ unique-string@1.0.0
│ │ │ └── crypto-random-string@1.0.0
│ │ ├─┬ write-file-atomic@2.4.2
│ │ │ ├── graceful-fs@4.1.15 deduped
│ │ │ ├── imurmurhash@0.1.4
│ │ │ └── signal-exit@3.0.2
│ │ └── xdg-basedir@3.0.0 deduped
│ ├─┬ gaxios@1.8.1
│ │ ├── abort-controller@2.0.2 deduped
│ │ ├── extend@3.0.2 deduped
│ │ ├── https-proxy-agent@2.2.1 deduped
│ │ └── node-fetch@2.3.0 deduped
│ ├── google-auth-library@3.1.0 deduped
│ ├── pumpify@1.5.1 deduped
│ └── stream-events@1.0.5 deduped
├─┬ hash-stream-validation@0.2.1
│ └─┬ through2@2.0.5
│   ├── readable-stream@2.3.6 deduped
│   └── xtend@4.0.1
├── mime@2.4.0
├─┬ mime-types@2.1.22
│ └── mime-db@1.38.0 deduped
├─┬ once@1.4.0
│ └── wrappy@1.0.2
├─┬ pumpify@1.5.1
│ ├── duplexify@3.7.1 deduped
│ ├── inherits@2.0.3 deduped
│ └─┬ pump@2.0.1
│   ├── end-of-stream@1.4.1 deduped
│   └── once@1.4.0 deduped
├── snakeize@0.1.0
├─┬ stream-events@1.0.5
│ └── stubs@3.0.0
├─┬ teeny-request@3.11.3
│ ├─┬ https-proxy-agent@2.2.1
│ │ ├─┬ agent-base@4.2.1
│ │ │ └─┬ es6-promisify@5.0.0
│ │ │   └── es6-promise@4.2.6
│ │ └─┬ debug@3.2.6
│ │   └── ms@2.1.1
│ ├── node-fetch@2.3.0
│ └── uuid@3.3.2
├─┬ through2@3.0.1
│ └── readable-stream@2.3.6 deduped
└── xdg-basedir@3.0.0
```
